code/ship/shipfx.cpp

```diff
@@ -3491,14 +3488,15 @@ int WE_Default::warpStart()
 		// now project the warp effect forward
 		vm_vec_scale_add( &pos, &pos, &objp->orient.vec.fvec, half_length );
 
-		// Effect time is 'SHIPFX_WARP_DELAY' (1.5 secs) seconds to start, warping_time 
-		// for ship to go thru, and 'SHIPFX_WARP_DELAY' (1.5 secs) to go away.
-		effect_time = warping_time + SHIPFX_WARP_DELAY + SHIPFX_WARP_DELAY;
+		// for arrivals, opening and closing times are both SHIPFX_WARP_DELAY
+		open_time = close_time = SHIPFX_WARP_DELAY;
 	}
 	else
 	{
-		compute_warpout_stuff(&effect_time, &pos);
-		effect_time += SHIPFX_WARP_DELAY;
+		// for departures, the opening time is the time it takes for the ship to actually approach the warp effect (not de jure, but de facto)
```

Comment:
"(not de jure, but de facto)" does not make sense.  You're saying "not rightfully so, but as it is because it is what it is".  But what is the rightful way to do it in the first place?

Reply:
What I meant was "this is probably not what Volition originally intended, but is how it ended up working".  In other words, I assume they intended for the opening and closing times to be `SHIPFX_WARP_DELAY` for all four possibilities (arrival open, arrival close, departure open, departure close).  But perhaps due to unintended side effects from implementation, the departure open time will always vary from ship to ship, depending on how long it takes that ship to move up to its departure point.

Reply:
I re-worded the comment.  See what you think.

Reply:
That is much clearer, thanks